chain/rust/src/builders/tx_builder.rs

```diff
@@ -227,10 +227,71 @@ fn min_fee(tx_builder: &TransactionBuilder) -> Result<Coin, TxBuilderError> {
 fn min_fee_with_exunits(tx_builder: &TransactionBuilder) -> Result<Coin, TxBuilderError> {
     let full_tx = fake_full_tx(tx_builder, tx_builder.build_body()?)?;
     // we can't know the of scripts yet as they can't be calculated until we build the tx
+
+    fn ref_script_orig_size_builder(utxo: &TransactionUnspentOutput) -> Option<(ScriptHash, u64)> {
+        utxo.output.script_ref().map(|script_ref| {
+            (
+                script_ref.hash(),
+                script_ref
+                    .raw_plutus_bytes()
+                    .expect("TODO: handle this")
+                    .len() as u64,
+            )
+        })
+    }
+    // let ref_script_orig_sizes: std::collections::BTreeMap<ScriptHash, u64> = if let Some(ref_inputs) = &tx_builder.reference_inputs {
+    //     ref_inputs.iter().chain(tx_builder.inputs.iter()).filter_map(ref_script_orig_size_builder).collect()
+    // } else {
+    //     tx_builder.inputs.iter().filter_map(ref_script_orig_size_builder).collect()
+    // };
+
+    // let mut total_ref_script_size = 0;
+    // for output in tx_builder.outputs.iter() {
+    //     if let Some(Credential::Script{ hash, .. }) = output.address().payment_cred() {
+    //         if let Some(orig_size) = ref_script_orig_sizes.get(hash) {
+    //             total_ref_script_size += *orig_size;
+    //             println!("USING REF SCRIPT {} | SIZE {}", hash.to_hex(), orig_size);
+    //         }
+    //     }
+    // }
+
+    let ref_script_orig_sizes: HashMap<ScriptHash, u64> =
+        if let Some(ref_inputs) = &tx_builder.reference_inputs {
+            ref_inputs
+                .iter()
+                .filter_map(ref_script_orig_size_builder)
+                .collect()
+        } else {
+            HashMap::default()
+        };
+
+    println!("\n\n\n ORIG SIZES:");
+    for (hash, size) in ref_script_orig_sizes.iter() {
+        println!("  orig {} - {}", hash.to_hex(), size);
+    }
+
+    let mut total_ref_script_size = 0;
+    for utxo in tx_builder.inputs.iter() {
+        if let Some(Credential::Script { hash, .. }) = utxo.output.address().payment_cred() {
+            println!("  ------ searching: {}", hash.to_hex());
```

Comment:
Probably we want to remove all these print statements?

Reply:
Yes. Sorry, I forgot to set it back to draft. I just wanted to check mergability.